dcos/errors.py

```diff
@@ -49,11 +49,18 @@ class DCOSAuthenticationException(DCOSHTTPException):
 
     :param response: requests Response object
     :type response: Response
+    :param message: An optional message for the exception. This can be useful
+                    to provide the user with more meaningful information.
+    :type message: str
     """
-    def __init__(self, response):
+    def __init__(self, response, message=None):
         self.response = response
```

Comment:
I wish we had a convention for composing messages from exceptions in the DC/OS CLI. I don't like the idea of just blindly passing in a message like this (without letting the exception itself supply some standard prefix for the message or something). It's probably OK for now, but we really need to architect this better in the future.

Also, can we reorganize this as:
```
def __init__(self, response, message=None):
    if message is None:
        message = "Authentication failed. Please run `dcos auth login`"

    self.response = response
    self.message = message
  		  
def __str__(self):
    return self.message
```

Finally, it seems strange to me to require a response to be sent in when we don't actually do anything with it.  I know you didn't add this, but it's weird nonetheless. I'd almost opt for removing it.